### PR TITLE
Feature/expenses model

### PR DESCRIPTION
**Pull request by MittchoBaroco**

### CODE REVIEW

File: app/models/expense.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+  # create a method call total that return a money object
+  monetize :total_cents
+
+  validate :total_greater_than_1_francs
+
+  def total_cents
+    return (training_cost_cents +
+    travel_cost_cents +
```
Comment (btihen) [inline app/models/expense.rb]: can you indent these on lines after the return - nicer style

File: spec/factories/expenses.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :expense do
+    training_cost { Faker::Number.between(100, 9000) }
```
Comment (btihen) [inline spec/factories/expenses.rb]: to clarify the money amounts use:
# 1CHF-90CHF
 training_costs_cents { Faker::Number.between(100, 9000) }
# or alternatively 
training_costs { Money.new(Faker::Number.between(1, 90), 'CHF') }

File: db/schema.rb [hunk 3]
```diff
@@ -31,6 +39,19 @@
     t.index ["department_head_id"], name: "index_departments_on_department_head_id"
   end
 
+  create_table "expenses", force: :cascade do |t|
+    t.integer "training_cost_cents", default: 0, null: false
+    t.string "training_cost_currency", default: "CHF", null: false
+    t.integer "travel_cost_cents", default: 0, null: false
```
Comment (btihen) [inline db/schema.rb]: defaut 0 null false - good
default 'CHF' null false - good

File: spec/models/expense_spec.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    it "invalidated expense that are less than 1 franc (100 cents)" do
+      expense = FactoryBot.build(:expense,
+        training_cost: 0,
+        travel_cost: 0,
+        accommodation_cost: 0,
+        dining_cost: 0
+      )
+
+      expect(expense).not_to be_valid
+      expect(expense.errors.messages[:base]).to include("total cannot be less than one")
```
Comment (btihen) [inline spec/models/expense_spec.rb]: please change to: 
include("total cannot be less than one Franc")

File: spec/factories/expenses.rb (hunk 2 above)
Comment (MittchoBaroco) [reply spec/factories/expenses.rb]: it's the reverse I think

File: app/models/expense.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    accommodation_cost_cents +
+    dining_cost_cents)
+  end
+
+  private
+
```
Comment (btihen) [inline app/models/expense.rb]: good call this is a good private method

File: spec/models/expense_spec.rb [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        accommodation_cost: 300.50,
+        dining_cost: 100.00
+      )
+    }
+
+    it "return correct total" do
+      # money object is instantiated with cents so 2700.50 = 270050
+      total_object = Money.new(270050)
+
+      expect(defined_expense.total).to eq(total_object)
```
Comment (btihen) [inline spec/models/expense_spec.rb]: this works? I see:
       def total_cents
, but I dont see:
       def total

File: app/models/expense.rb [hunk 7]
```diff
@@ -27,9 +27,10 @@ class Expense < ApplicationRecord
 
   def total_cents
     return (training_cost_cents +
-    travel_cost_cents +
-    accommodation_cost_cents +
-    dining_cost_cents)
+      travel_cost_cents +
+      accommodation_cost_cents +
+      dining_cost_cents
+    )
```
Comment (btihen) [inline app/models/expense.rb]: thanks

**Pull request merged**